graph/src/components/store.rs

```diff
@@ -397,19 +415,255 @@ pub trait Store: Send + Sync + 'static {
+                EntityFilter::new_in("currentVersion", version_ids.clone()),
+                EntityFilter::new_in("pendingVersion", version_ids.clone()),
+            ])))?;
+        let subgraph_ids_with_version_as_current_or_pending =
+            subgraphs_with_version_as_current_or_pending
+                .iter()
+                .map(|subgraph_entity| subgraph_entity.id().unwrap())
+                .collect::<HashSet<_>>();
+        ops.push(EntityOperation::AbortUnless {
+            description: "Same set of subgraphs must have these versions as current or pending"
+                .to_owned(),
+            query: SubgraphEntity::query().filter(EntityFilter::Or(vec![
+                EntityFilter::new_in("currentVersion", version_ids.clone()),
+                EntityFilter::new_in("pendingVersion", version_ids),
+            ])),
+            entity_ids: subgraph_ids_with_version_as_current_or_pending
+                .into_iter()
+                .collect(),
+        });
+
+        // Produce summaries, deriving flags from information in subgraph entities
+        let version_summaries =
+            versions
+                .into_iter()
+                .map(|version_entity| {
+                    let is_current = subgraphs_with_version_as_current_or_pending.iter().any(
+                        |subgraph_entity| {
+                            subgraph_entity.get("currentVersion")
+                                == Some(version_entity.get("id").unwrap())
+                        },
+                    );
+                    let is_pending = subgraphs_with_version_as_current_or_pending.iter().any(
+                        |subgraph_entity| {
+                            subgraph_entity.get("pendingVersion")
+                                == Some(version_entity.get("id").unwrap())
+                        },
+                    );
+
+                    SubgraphVersionSummary {
+                        id: version_entity.id().unwrap(),
+                        subgraph_id: version_entity
+                            .get("subgraph")
+                            .unwrap()
+                            .to_owned()
+                            .as_string()
+                            .unwrap(),
+                        deployment_id: SubgraphDeploymentId::new(
+                            version_entity
+                                .get("deployment")
+                                .unwrap()
+                                .to_owned()
+                                .as_string()
+                                .unwrap(),
+                        )
+                        .unwrap(),
+                        current: is_current,
+                        pending: is_pending,
+                    }
+                })
+                .collect();
+
+        Ok((version_summaries, ops))
+    }
+
+    /// Produce the EntityOperations needed to create/remove SubgraphDeploymentAssignments to
```

Comment:
Comment lines are too long, I think.

Reply:
Oh what's our preferred max width? I've been wrapping to 100

Reply:
Whatever rustfmt does to the code. 90, I think?

Reply:
Sure. I tightened these up, and maybe next week we can turn on auto comment wrapping in rustfmt